pinns/src/sysctl.c

```diff
@@ -0,0 +1,82 @@
+#define _GNU_SOURCE
+
+#include <glib.h>
+#include <string.h>
+#include <fcntl.h>
+
+#include "sysctl.h"
+#include "utils.h"
+
+static int separate_sysctl_key_value (char* sysctl_key_value, char* sysctl_key, char* sysctl_value);
+static int write_sysctl_to_file (char * sysctl_key, char* sysctl_value);
+
+int configure_sysctls (const GPtrArray *sysctls) {
+  size_t n_sysctls = 0;
+  while (g_ptr_array_index (sysctls, n_sysctls))
+  {
+    char* key = NULL;
+	char* value = NULL;
+    if (separate_sysctl_key_value (g_ptr_array_index (sysctls, n_sysctls++), key, value) < 0)
+      return -1;
+    if (write_sysctl_to_file(key, value) < 0)
+      return -1;
+  }
+
+  return 0;
+}
+
+static int separate_sysctl_key_value (char* sysctl_key_value, char* sysctl_key, char* sysctl_value)
+{
+  char * equals_token = strchr (sysctl_key_value, '=');
+  if (!equals_token) {
+	pwarnf("sysctl must be in the form of 'key=value'; '=' missing");
+    return -1;
+  }
+  *equals_token = '\0';
+  sysctl_key = sysctl_key_value;
+  sysctl_value = equals_token+1;
+  if (!strlen (sysctl_key))
+  {
+	pwarnf ("sysctl must be in the form of 'key=value'; key is empty");
+    return -1;
+  }
+  if (!strlen (sysctl_value))
+  {
+	pwarnf ("sysctl must be in the form of 'key=value'; key is value");
+    return -1;
+  }
+  return 0;
+}
+
+static int write_sysctl_to_file (char * sysctl_key, char* sysctl_value)
```

Comment:
todo: give @giuseppe credit for this function

Reply:
could we use directly libcrun or is it overkill?  It doesn't need to be linked and it could just be a git submodule

Reply:
It is kind of overkill. All the autoconf/automake configuration for a 15 line function ultimately doesn't seem worth it.

The interaction with libcrun was also awkward, because we need to call pinns before we write the `config.json` to disk. To get the sysctls from it, we needed to write a temporary one, then use it to configure the namespaces, then use those namespaces to configure the real one.

Ultimately, I like this flow better.

Reply:
I thought we could use directly the function without going through the `config.json` part.

But I agree autoconf/automake for a function is overkill :-)